### DISCUSSION TITLE
Can't delete folder .next

### DISCUSSION THREAD

**Discussion by IRediTOTO** [vercel/next.js#14394] [Help]
# Bug report

## Describe the bug

Hi, when I run: `npm run build` my app get an error, it's ok, then I fixed the error
But when I run `npm run dev ` , I got this error
```
[Error: ENOTEMPTY: directory not empty, rmdir 'D:\Job\my-shop\.next\server\static\development\pages\api\item'] {
  errno: -4051,
  code: 'ENOTEMPTY',
  syscall: 'rmdir',
  path: 'D:\\Job\\my-shop\\.next\\server\\static\\development\\pages\\api\\item'
}
```
I tried many ways to delete .next folder, but not work, finally I must login to Safe mode of window to delete folder. 
I guest it come from name folder ? I set folder name is : api/item/[itemId]  

## System information

- OS: Windows 10
- Version of Next.js: 9.4
- Version of Node.js: 12.x

**Comment by IRediTOTO** [comment]
I tried to delete with powershell 
```
PS C:\WINDOWS\system32>  Remove-Item -path D:\Job\my-shop\.next -force

Confirm
The item at D:\Job\my-shop\.next has children and the Recurse parameter was not specified. If you continue, all
children will be removed with the item. Are you sure you want to continue?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): A
Remove-Item : Access to the path 'D:\Job\my-shop\.next\server\static\development\pages\api\item\[itemId]' is denied.
At line:1 char:2
+  Remove-Item –path D:\Job\my-shop\.next -force
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    **+ CategoryInfo          : PermissionDenied: (D:\Job\my-shop\.next:String) [Remove-Item], UnauthorizedAccessExce
   ption
    + FullyQualifiedErrorId : RemoveItemUnauthorizedAccessError,Microsoft.PowerShell.Commands.RemoveItemCommand**
```

**Comment by IamLizu** [comment]
You are getting this error because the `.next` directory is **not empty**. But wait, what you actually did to fix your app error which you got during `npm run build`? 

Usually, the `npm run dev` or `build` commands handles everything when it comes to re-generate new files, did you manually create any directory in `./next`?

**Reply by IRediTOTO** [reply]
Hi, I ran` npm run build` and just got my code logic error, it's easy to fix.
Then I tried run `npm run dev` to test code and it can't run because .next can't be deleted to create new one. 
I didn't touch to .next folder
This is weird error, It can come from anywhere: my SSD error, next error,... Just post here to find someone got this problem too.

**Reply by IamLizu** [reply]
Let's try a workaround, create `next.config.js` in your project's root and set it as bellow, if you already have a file, just add `distDir: 'build'` to it. Now build your app, this time your build will be saved in `build` directory in your project root. See if your issue still exists.
```
module.exports = {
  distDir: 'build',
}
```

**Reply by estyh** [reply]
I was getting an error running npm run build, following the next official basics tutorial, error was ENOTEMPTY: directory not empty - on the .next dirs... adding the .next.config.js file and the contents above fixed it.
I'm on windows 10

**Reply by nguyentri-art** [reply]
Hello ,Everytime i'm open VScode and try to `npm run dev `   again, this happen ,i'm try to fix this and this fix someway , but like i say , everytime i'm open project and countinue , this error again ,this is so annoying , someone help me , im work with` next.js` . on Windows 10 to

**Reply by GabOsaz** [reply]
Yes, adding 
module.exports = {
  distDir: 'build',
} 
fixed it for me too

**Reply by DevKaliper** [reply]
Yeah, i was facing that problem too. Im using react with Nextjs 14 so the new next.config.js file changed a little. Now its named with .mjs extension -> next.config.mjs

const nextConfig = {distDir: 'build'};

export default nextConfig;

**Comment by brunocrosier** [comment]
I was still getting this issue - I think it might be related to the fact that the folder is on my `D:\` drive instead of the default `C:\`

the `next.config.js` fix worked but would be good if it could work out of the box!

**Comment by kelvinsanchez15** [comment]
In my case closing VS Code allows me to delete the .next folder but looking for another alternative I realized that the problem was generated by the Prisma extension, I disabled it and with that my problem was solved.

I opened an Issue describing the bug on their repository: https://github.com/prisma/language-tools/issues/606

**Reply by brunocrosier** [reply]
Aaaah good shout @kelvinsanchez15 I have that extension running too!

**Reply by kelvinsanchez15** [reply]
Nice, we just have to wait for a fix from them.

**Reply by MalvinJay** [reply]
I suddenly started facing this issue. A quick and unpleasant fix is to delete the node_modules folder and re-install packages again

**Comment by mikenath223** [comment]
You could also delete the `.next` folder directly from vscode terminal by running this command: `rm -rf .next`

**Reply by baxa29102001** [reply]
Thank you bro

**Reply by bryanprimus** [reply]
`rm: cannot remove '.next/trace': Permission denied` not working for me

**Comment by baxa29102001** [comment]
I have come across such problems after npm run dev and i delete my .next folder

**Comment by JosXa** [comment]
Leaving `next dev` running while switching branches seems to be causing this a lot of times... no chance to delete the folder on Windows whatsoever, even after closing the IDE and shutting down all apparent processes using it.

When this occurs, it will also cause the dev command to hang right here without any additional info:
![chrome_CUdfTZkbPj](https://user-images.githubusercontent.com/7313176/150702882-a42194fc-b0be-4c82-9e4b-f04803cd5687.png)

**Comment by ninja-** [comment]
I had the same problem and was able to fix it with `taskkill /im node.exe /f`
What exactly process was leaking after completely closing VSCode and all dev terminal, I have no idea...

**Reply by ziadevcom** [reply]
👆🏾 This here is what worked for me. Thanks buddy.

**Reply by justatestsara** [reply]
thank you! :D

**Comment by abhishekprajapati1** [comment]
If anyone still facing this issue. Let me tell you, if you have any script running in terminal such as `next build` or `next dev`,  you can't delete .next folder because it contains .trace file. So first make sure to eliminate all scripts running in terminal and then you can delete .next directory along with its content.

**Reply by ali-raza-166** [reply]
I was trying to run `next build` while `next dev` was already running in a separate terminal. I was unable to run the build command with the exception `uncaughtException [Error: EPERM: operation not permitted, open 'F:\NextJS\...`. So, I came across your answer. I eliminated the `next dev` script and tried to build it, and it did work.  Thanks champ

**Comment by yuriyPetrushyn** [comment]
if you have messages that you do not have permission try to enter before 'sudo' - sudo rm -rf foldername/

**Comment by lachlansleight** [comment]
This error has literally been plaguing me for years. Next will crash for some reason (usually because I'm an idiot and set up some infinite loop). I'll quit the process in my terminal, and be unable to restart the server because of the trace file within .next. 

-  Closing VSCode doesn't help
- Killing the node process doesn't help 
- Running `Remove-File -Force trace` in PowerShell (as admin) doesn't work (I get the UnauthorizedAccessException reported above). File locksmith doesn't report the file being in use

The file tends to be quite large (today's is 9.2MB), but opening it up in any kind of editor shows an empty file. I've never encountered such an obtuse file in any other computing context throughout my whole life!

The only way I've ever been able to solve this is by restarting my PC, which is just...not my favourite thing in the world. Could next perhaps be modified to maybe not do such weird shit when creating this trace file?

**Comment by thienlovedev** [comment]
I have a problem, i hope get a help
I have a file with name is 'loading.jsx' inside my component ( app/mycomponent/loading/jsx)
when i deleted that file , an error showing up "The system cannot find the file specified. (os error 2)" although i didn't import any files or anything related that ! It seem has an error in file build of nextjs
Any body know that problem
Sorry my english is not good, i hope you ignore
Thank a lot !
![file](https://github.com/vercel/next.js/assets/133108424/96bb6b3e-e4e8-4413-8bfb-f8112af8bb8d)

**Comment by FJRG2007** [comment]
I have the same problem trying to delete the file
